projects/core/src/lib/components/toolbar/clicked-coordinates/clicked-coordinates.component.ts

```diff
@@ -0,0 +1,85 @@
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { concatMap, filter, Observable, Subject, takeUntil, tap } from 'rxjs';
+import { MapClickToolConfigModel, MapClickToolModel, MapService, ToolTypeEnum } from '@tailormap-viewer/map';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Clipboard } from '@angular/cdk/clipboard';
+import { $localize } from '@angular/localize/init';
+import { Store } from '@ngrx/store';
+import { isActiveToolbarTool } from '../state/toolbar.selectors';
+import { registerTool, toggleTool } from '../state/toolbar.actions';
+import { ToolbarComponentEnum } from '../models/toolbar-component.enum';
+
+@Component({
+  selector: 'tm-clicked-coordinates',
+  templateUrl: './clicked-coordinates.component.html',
+  styleUrls: ['./clicked-coordinates.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ClickedCoordinatesComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+  public toolActive$: Observable<boolean>;
+
+  constructor(
+    private store$: Store,
+    private mapService: MapService,
+    private snackBar: MatSnackBar,
+    private clipboard: Clipboard,
+  ) {
+    this.toolActive$ = this.store$.select(isActiveToolbarTool(ToolbarComponentEnum.SELECT_COORDINATES));
+  }
+
+  public ngOnInit(): void {
+    this.mapService.createTool$<MapClickToolModel, MapClickToolConfigModel>({
+      type: ToolTypeEnum.MapClick,
+    })
+      .pipe(
+        takeUntil(this.destroyed),
+        filter(Boolean),
+        tap(clickTool => {
+          this.store$.dispatch(registerTool({ tool: { id: ToolbarComponentEnum.SELECT_COORDINATES, mapToolId: clickTool.id }}));
+        }),
+        concatMap(clickTool => clickTool.mapClick$),
+      )
+      .subscribe(mapClick => {
+        this.handleMapClick(mapClick);
```

Comment:
See my comments below. Instead of subscribing here and execute this method it would be a better approach to chain the Observables using concatMap/switchMap, so after `concatMap(clickTool => clickTool.mapClick$),` you could add another `concatMap` to convert the coordinates to readable coordinates using the `this.mapService.getUnitsOfMeasure$()` Observable, for example:

```
concatMap(mapClickEvt => {
  return this.mapService.getUnitsOfMeasure$()
    .pipe(map(unit => `${evt.mapCoordinates[0].toFixed(decimals)}, ${evt.mapCoordinates[1].toFixed(decimals)}`));
}),
concatMap(coordinates => {
  const coordinatesMsg = $localize`Selected coordinates: ${coordinates}`;
  return this.snackBar.open(coordinatesMsg, $localize`Copy`).onAction();
}),
```
and then in the subscribe you end up with the 'Copy' button action the user executed so there you could do this code:
```
if (this.clipboard.copy(coordinatesMsg)) {
  this.snackBar.open($localize`Success`, '', {duration: 5000});
} else {
  this.snackBar.open($localize`Failed to copy to clipboard`, '', {duration: 5000});
}
```